### ISSUE TITLE
Zoom to selection fails

### ISSUE THREAD

**Issue by katterfelto** [wfletcher/EasyGIS.NET#130]
I have a shapefile in EPSG:27700 (GB NGR). When a tiled base layer is loaded (e.g. Open Street Maps) `SFMap.ZoomToSelection()` fails to zoom the the correct coordinates.

My initial solution:

The extent that is compiled from the selected objects in each shapefile needs to be transformed to EPSG:3857 before the call to `SFMap.FitToExtent(RectangleD extent)` is made.

The following diff shows my change, is there a better way?

```
diff --git a/EGIS.Controls/SFMap.cs b/EGIS.Controls/SFMap.cs
index b09fe3e..a0edef1 100644
--- a/EGIS.Controls/SFMap.cs
+++ b/EGIS.Controls/SFMap.cs
@@ -1047,20 +1047,35 @@ namespace EGIS.Controls
                 System.Collections.ObjectModel.ReadOnlyCollection<int> selectedIndicies = layer.SelectedRecordIndices;
                 if (selectedIndicies.Count > 0)
                 {
+                    RectangleD selExtent = RectangleD.Empty;
+                    bool selExtentSet = false;
                     for (int n = 0; n < selectedIndicies.Count; ++n)
                     {
-                        if (extentSet)
+                        if (selExtentSet)
                         {
-                            extent = RectangleD.Union(extent, layer.GetShapeBoundsD(selectedIndicies[n]));
+                            selExtent = RectangleD.Union(selExtent, layer.GetShapeBoundsD(selectedIndicies[n]));
                         }
                         else
                         {
-                            extent = layer.GetShapeBoundsD(selectedIndicies[n]);
-                            extentSet = true;
+                            selExtent = layer.GetShapeBoundsD(selectedIndicies[n]);
+                            selExtentSet = true;
                         }
                     }
+
+                    if (extentSet)
+                    {
+                        extent = RectangleD.Union(extent,
+                            selExtent.Transform(layer.CoordinateReferenceSystem, this.MapCoordinateReferenceSystem));
+                    }
+                    else
+                    {
+                        extent = selExtent.Transform(layer.CoordinateReferenceSystem,
+                            this.MapCoordinateReferenceSystem);
+                        extentSet = true;
+                    }
                 }
             }
             if (extentSet) FitToExtent(extent);
         }
```

**Comment by katterfelto** [comment]
The way I tested this:

- First add a tool bar button to `egis.exe` to trigger a Zoom To Selection call. Compile and Run.
- Select `Options->Use CRS From First ...`.
- Load a shapefile (with a CRS which is not EPSG:3587).
- Select the Open Street Map base layer.
- Set the `IsSelectable` render setting to `True`.
- Now select some of the elements.
- Finally click the zoom to selection button.

At this point:
- Without the change, the view will switch to a different area of the globe and not display the selected items (or crash depending on the shapefile coordinate values).
- With the code change the selected items will be displayed with the base layer correctly shown below them.

**Comment by wfletcher** [comment]
fixed in PR#131
